devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -312,18 +312,17 @@ def zwaveEvent(DoorLockOperationReport cmd) {
 			result << response(secure(zwave.associationV1.associationGet(groupingIdentifier:1)))
 		}
 	}
-	result = result ? [createEvent(map), *result] : createEvent(map)
 	if (generatesDoorLockOperationReportBeforeAlarmReport()) {
 		// we're expecting lock events to come after notification events, but for specific yale locks they come out of order
-		runIn(3, "delayLockEvent", [data: [event: result]])
+		runIn(3, "delayLockEvent", [data: [map: map, result: result]])
         return [:]
 	} else {
-		return result
+		return result ? [createEvent(map), *result] : createEvent(map)
 	}
 }
 
 def delayLockEvent(data) {
-	sendEvent(data.event)
+	sendEvent(data.result ? [createEvent(data.map), *data.result] : createEvent(data.map))
```

Comment:
Now theoretically this looks correct. But today I tried this out by self publishing the DTH and `delayLockEvent` doesn't get called. I made `generatesDoorLockOperationReportBeforeAlarmReport` return true explicitly and `delayLockEvent` gets scheduled to be called in 3 seconds but it doesn't get called and as a result event doesn't get generated. May be I am missing something or something is broken in production server because of which `delayLockEvent` is not getting called. Either way, @greens can you try executing this in your account by self publishing and check whether `delayLockEvent` gets called? By the way does DTH support calling a function specified in runIn? I know scheduling works in smartapps but I am not sure about DTH so please confirm this once.

Reply:
I tested on my account with logging set up with the correct device and delayLockEvent was being called for me. 

Try testing it in the IDE with some additional logging statements. Because the event _shouldn't_ show up.

Reply:
I added a log in `delayLockEvent` and it didn't print. Anyway, if it's working for you then it's Ok.